packages/wrangler/src/versions/upload.ts

```diff
@@ -50,6 +50,7 @@ type Props = {
 	env: string | undefined;
 	compatibilityDate: string | undefined;
 	compatibilityFlags: string[] | undefined;
+	experimentalAssets: string | undefined;
```

Comment:
```suggestion
	experimentalAssets: ExperimentalAssets | undefined;
```
since `wrangler.toml` supports
```
[experimental_assets]
directory = "dist"
binding = "binding_name"
```

Reply:
I copied this from `https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/deploy/deploy.ts#L79` and it's passed just the `.directory` property on https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/deploy/index.ts#L372

Happy to change it in both places if you'd prefer?

Reply:
Since this is the new experimental assets, there is nothing to be done with the binding name yet. So I think it is OK to match deploy.ts here and @emily-shen can update it when she implements the new assets support.

Reply:
ah, I see. I think `string` was temporarily used there as a placeholder, until the type of `experimental _assets` was fully settled on (there's been some back and forth around it). We've definitely settled on `ExperimentalAssets` since then. The reason why `deploy` only gets `experimentalAssets.directory` passed in, is probably because of a similar reason. The code around experimental assets is very much in flux atm, which is why you're seeing these inconsistencies & rough edges. I've already worked through these for `dev` and Emily is currently busy with the `deploy` bit, so it'll all be smoothed out soon. Fantastic catch tho!

So to tl;dr:
- use the `ExperimentalAssets` type for all experimental assets things (the **ONLY** exception to that is `getEntry()`, which takes in `experimentalAssets: string` see my comment [here](https://github.com/cloudflare/workers-sdk/pull/6370/files#diff-524c0b6c44e4e63168f1849abe9420fbb7c44f64c3a6166f44515fdd00b2118dR207))
- pass the entire object to `deploy` (Emily is taking care of this)/`versions` etc